python/pyspark/sql/streaming/value_state_client.py

```diff
@@ -40,12 +40,15 @@ def exists(self, state_name: str) -> bool:
         status = response_message[0]
         if status == 0:
             return True
-        elif status == 1:
-            # server returns 1 if the state does not exist
+        elif status == 1 and "doesn't exist" in response_message[1]:
```

Comment:
I'd recommend to use the different status code instead of parsing. Please consider the change relying on string/hardcode to be unacceptable except specific needs.